concordium-consensus/src/Concordium/External.hs

```diff
@@ -1000,381 +995,16 @@ toCString lbs = do
-getConsensusStatus cptr = jsonQuery cptr Q.getConsensusStatus
-
--- ** Queries against latest tree
-
--- | Returns a null-terminated string with a JSON representation of the current branches from the
---  last finalized block (inclusive).
-getBranches :: StablePtr ConsensusRunner -> IO CString
-getBranches cptr = jsonQuery cptr Q.getBranches
-
--- | Get the list of live blocks at a given height.
---  The height is interpreted relative to the genesis block at the specified index.
---  The last parameter indicates whether to restrict to a single genesis (if it is a non-zero value).
---  Returns a null-terminated string encoding a JSON list.
-getBlocksAtHeight ::
-    StablePtr ConsensusRunner ->
-    -- | Block height to query
-    Word64 ->
-    -- | Genesis index that block height is based on
-    Word32 ->
-    -- | Non-zero to restrict to blocks at specified genesis index
-    Word8 ->
-    IO CString
-getBlocksAtHeight cptr height genIndex restrict =
-    jsonQuery cptr $
-        fmap fst
-            <$> Q.getBlocksAtHeight
-                (BlockHeight height)
-                (GenesisIndex genIndex)
-                (restrict /= 0)
-
--- | Retrieve the last finalized block height relative to the most recent genesis index. Used for
--- resuming out-of-band catchup.
-getLastFinalizedBlockHeight ::
-    StablePtr ConsensusRunner ->
-    IO Word64
-getLastFinalizedBlockHeight cptr = do
-    (ConsensusRunner mvr) <- deRefStablePtr cptr
-    theBlockHeight <$> runMVR Q.getLastFinalizedBlockHeight mvr
-
 getNumberOfNonFinalizedTransactions :: StablePtr ConsensusRunner -> IO Word64
 getNumberOfNonFinalizedTransactions cptr = do
     (ConsensusRunner mvr) <- deRefStablePtr cptr
     fromIntegral <$> runMVR Q.getNumberOfNonFinalizedTransactions mvr
 
--- ** Block-indexed queries
-
--- | Given a null-terminated string that represents a block hash (base 16), returns a null-terminated
---  string containing a JSON representation of the block.
---  If the block hash is invalid or unknown, this returns the JSON null value.
---  For details of the value returned, see 'Concordium.Queries.Types.BlockInfo'.
-getBlockInfo :: StablePtr ConsensusRunner -> CString -> IO CString
-getBlockInfo cptr blockcstr =
-    decodeBlockHash blockcstr >>= \case
-        Nothing -> jsonCString AE.Null
-        Just bh -> jsonQuery cptr (Q.responseToMaybe <$> Q.getBlockInfo (Queries.Given bh))
-
--- | Get the list of transactions in a block with short summaries of their effects.
---  Returns a null-terminated string encoding a JSON value.
---  If the block hash is invalid or unknown, this returns the JSON null value.
---  For details of the value returned, see 'Concordium.Queries.Types.BlockSummary'.
-getBlockSummary :: StablePtr ConsensusRunner -> CString -> IO CString
-getBlockSummary cptr blockcstr =
-    decodeBlockHash blockcstr >>= \case
-        Nothing -> jsonCString AE.Null
-        Just bh -> jsonQuery cptr (Q.getBlockSummary bh)
```

Comment:
Also remove `Queries.getBlockSummary`.

Reply:
Also the `BlockSummary` type can (probably) be removed from base.

Reply:
Yes.